### DISCUSSION TITLE
inserting link between the nodes of two different .mm file

### DISCUSSION THREAD

**Discussion by edbeee** [freeplane/freeplane#2071] [Help]
Hi there! I wonder how can I insert link between the two nodes of two different .mm file?  If I click on the link of the one node it should lead to the node of the other .mm file that I connected.
For example, in the following screenshot display. there are two mindmaps. I want to insert link on the highlighted node of the left mindmap, and the link should lead me to the highlighted node of the second(right) mindmap. How can I do it?
![image](https://github.com/user-attachments/assets/c7949d43-f3eb-4bd7-9721-4aec1b5bc0c0)

Which option will allow me to do that?
![image](https://github.com/user-attachments/assets/26745d05-7dcd-41df-87e8-b37a8429909e)

**Comment by i-plasm** [comment]
Hello edbeee,

1. The fastest way is using link anchors. You can use the options "Set link anchor" and "Make link to / from link anchor". These are found in the `Insert->Link` menu. I think the functionalities are self-explanatory, but in case you'd like an example: https://sourceforge.net/p/freeplane/discussion/758437/thread/1f241891/#0d44
   - If you need the link to go both ways, you can use this script by macmarrum (you need to set the anchor beforehand, and run it on target node): https://github.com/freeplane/freeplane/discussions/1255#discussioncomment-6246355
1. You can also do it manually, by copying and pasting the node URI. Right click on the node you want to link to and choose `Copy -> Copy node URI`. Then go to the node where you want to insert the hyperlink, and `Insert->Link->Add hyperlink (URL)…`  (you can acess it by right click, or if you're on Windows you can use the `Ctrl+K` hot key as well)

**Reply by edbeee** [reply]
Thank you @i-plasm . Your answer perfectly helped me to do the link.

However, I wonder, what are the pros and cons of using anchor(solution 1) or manual linking (solution 2)?

In addition, in future If I change my location of my mindmaps(unintentionally) which solution would create an issue?

**Reply by i-plasm** [reply]
I'm glad I could help

> However, I wonder, what are the pros and cons of using anchor(solution 1) or manual linking (solution 2)?

Solution 1 reads the `Preferences…->Environment->Hyperlink types->Links` preference, and it will produce a relative or absolute link accordingly. So, it is more flexible than solution 2, which only creates absolute links.

One advantage I see of solution 2 is that the links have the `freeplane:` scheme, so if you export your map to html etc, then you'll be able to browse to the linked node from external apps. 

> In addition, in future If I change my location of my mindmaps(unintentionally) which solution would create an issue?

For being able to move maps with relative freedom, you'd need solution 1, and also that you commit to 1) always maintaining your linked & linking maps in the same relative-to-each-other folder structure, and 2) making sure `Preferences…->Environment->Hyperlink types->Links` is set to `Relative` in every freeplane installation where you edit your maps.

Solution 2 always creates absolute links, so every time you move a linked map, you have to perform a search and replace with an external text editor on the other map to update the links (the replacement URI must be encoded, and one way to obtain it is via `Edit->Copy->Copy node URI`, carefully eliminating the `#ID_12345...` part). Always back up your map before such operation - it can be error-prone if not done carefully.

So, both have pros and cons.

**Reply by edbeee** [reply]
I seeeee!. Very clear explanation, now I understand. Thank you❤ @i-plasm.
